### PR TITLE
Custom NewRelic logger

### PR DESCRIPTION
**Pull request by SquarePlayn**
## Proposed changes
Adds a custom logger for logging to NewRelic, including various default fields.

Related to [AB#42](https://dev.azure.com/jdokter/973dd796-5b61-4e4a-a52b-ab66f88cef8a/_workitems/edit/42)

## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Unit testing
None specifically for this.

### Manual testing
Use the logger in a repository (see README.md on how to activate it) and ensure it works and is robust.

## Further comments

### CODE REVIEW

File: src/Logging/NewRelicLogger.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            'version' => Config::has('app.version') ? config('app.version') : 'Not implemented',
+            'name' => config('app.name'),
+            'hostname' => gethostname(),
+            'url' => config('app.url'),
+        ];
+
+        // Add info about the Auth0 user performing the request we are running (if any)
+        $record['user'] = [
+            'authenticated' => optional(request())->user_id,
+            'user_id' => optional(request())->user_id,
```
Comment (marijnvanderhorst) [inline src/Logging/NewRelicLogger.php:50]: These two values are the same?

Comment (SquarePlayn) [reply src/Logging/NewRelicLogger.php]: Fixed. The 'authenticated' field is only supposed to show if the request is authenticated at all (as a boolean)

File: src/Logging/NewRelicLogger.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    /**
+     * Add extra metadata to every request
+     */
+    public function includeMetaData(array $record): array
+    {
+        // Include info about which service this is
+        $record['app'] = [
+            'repository' => Config::has('app.repository') ? config('app.repository') : 'Not implemented',
+            'version' => Config::has('app.version') ? config('app.version') : 'Not implemented',
```
Comment (marijnvanderhorst) [inline src/Logging/NewRelicLogger.php:41]:
```suggestion
            'repository' => config('app.repository', 'Not implemented'),
            'version' => config('app.version', 'Not implemented'),
```

**Review by marijnvanderhorst — DISMISSED** [review]
Looks good, 1 suggested change

**Review by marijnvanderhorst — APPROVED** [review]

**Pull request merged**